// paysafe/index.js
var request = require('request');
var pay_safe_auth_api = "https://api.test.paysafe.com/cardpayments/v1/accounts/1001267670/auths";
var pay_safe_verify_api = "https://api.test.paysafe.com/cardpayments/v1/accounts/1001258970/verifications";
var pay_safe_refund_api = "https://api.test.paysafe.com/cardpayments/v1/accounts/1001267670/settlements/settlementid/refunds";
var pay_safe_single_apiKey= "T1QtMjQ1NTEwOkItcWEyLTAtNWJhMTJiZmYtMC0zMDJjMDIxNDJlMDJhN2VlNTJiZWNiOTZkZDRkODIzYWEwMzJiYjBjMjM1ZWQ2ZGYwMjE0NWFlYzljMWRkZTA1ZTg5YjYzZjg2YzNkZTI3YzVkMjhhMWUxYTgyMA==";
var pay_safe_enrollment_api = "https://api.test.paysafe.com/threedsecure/v1/accounts/1001267670/enrollmentchecks";
var pay_safe_enrollment_auth = "https://api.test.paysafe.com/threedsecure/v1/accounts/1001267670/enrollmentchecks/enrollment_id/authentications";
var pay_safe_username = "test_technical";//pmle-239490
var pay_safe_password = "B-qa2-0-5ba126e5-0-302d02150082dccd8f2b32b724bf5c01d203fcfd6c5f1d8fd1021448b23b3b0424c1642c2a77507b892a493faa324c";
var post_fields = '';

exports.createPaysafePayment = function(para1,callback) {
	var api_url =  pay_safe_auth_api;
	var username = pay_safe_username;
	var password = pay_safe_password;
	
	
	var headers = {
    'Content-type': 'application/json',
    'Accept': 'application/json'
	};
	
	var dataString = post_fields;
	var options = {
	    url: api_url,
	    method: 'POST',
	    headers: headers,
	    body: dataString,
	    auth: {
	        'user': username,
	        'pass': password
	    }
	};
	
	request(options, function callback(error, response, body) {  
	    if (!error && response.statusCode == 200) {
	        console.log(body);
	        var output_response = {};
	        var paysafe_api_resp = JSON.parse(body);
	        var error_msg = paysafe_api_resp.error;
			if(error_msg !="" ) {
				output_response.status = "fail";
				output_response.msg = paysafe_api_resp;
				callback(null,output_response);
			} else {
				output_response.status = "success";
				output_response.msg = paysafe_api_resp;
				callback(null,output_response);	
			}
	    } else {
	    	callback(error, null);	
	    }
	});
	
	
}
